compiler/shared_ast/optimizations.ml

```diff
@@ -100,61 +100,17 @@ let rec optimize_expr :
        the matches and the log calls are not preserved, which would be a good
        property *)
     match Mark.remove e with
-    | EApp
-        {
-          f =
-            ( EOp { op = Not; _ }, _
-            | ( EApp
-                  {
-                    f = EOp { op = Log _; _ }, _;
-                    args = [(EOp { op = Not; _ }, _)];
-                  },
-                _ ) ) as op;
-          args = [e1];
-        } -> (
+    | EAppOp { op = Not; args = [(ELit (LBool b), _)]; _ } ->
```

Comment:
OK I see, because the logging call wraps the `EAppOp`, you no longer need to match on it. Clever!